### ISSUE TITLE
Unified Screen Locking Capability

### ISSUE THREAD

**Issue by theblu3j** [deskflow/deskflow#7902]
### Feature Request

~~When I have my laptop (client) and main PC (server) together using Deskflow, it'd be really great if it was possible to have keybinds that triggered on both server and clients. For example, when I leave my desk I use Win+L to lock my devices (on KDE, as far as I can tell it's the same keybind for Windows as well). It'd be nice to be able to lock both at once. Could also have other uses such as running a script on both computers (start/stop syncthing or some other service maybe).~~

EDIT: Aaaand I've just had another look through the settings and saw there is unified keybinds of some sort, but not possible to use the Win+L keybind. Probably possible to use a workaround, but it'd be nice to have a built in solution for synchronized screen locking regardless. Updating the title now.

**Comment by theblu3j** [comment]
One note, synchronized screen unlocking would likely be an enormous pain to do in a secure way across all different variations of OS but I'm mostly concerned with screen locking.

**Comment by nbolton** [comment]
Yeah, I think syncronised screen locking would be fairly safe. Perhaps we could implement a new hotkey mode that can broadcast different keystroks to all computers at the same time. It may need to be a hotkey key combo on the server other than the actual lock key (e.g. IIRC, Windows blocks us from intercepting Win+L for security).

If someone discovered an exploit to this, the worst case would be a DoS where they could spam the lock key broadcast and permanently lock you out of all computers... come to think of it that would be quite annoying. Then again, if someone discovered how to exploit broadcasting of keys, they could just spam a bunch of random input, so it's similar risk as now.

Other other hand, like you said, this should be lock-only, since unlock would be a big security vulnerability (if it's even possible).

**Comment by sithlord48** [comment]
You can do this using the current hotkey setup .. i would ask the OP to explore them.

**Comment by nbolton** [comment]
> You can do this using the current hotkey setup .. i would ask the OP to explore them.

The problem could be more nuanced than you are imagining. You can broadcast the same keystroke to multiple clients but AFAIK, you cannot broadcast different keystrokes to different clients with one keystroke on the server. This would be necessary if the clients are different operating systems, which they usually are.

![image](https://github.com/user-attachments/assets/dd903c96-72e2-4dce-901b-e31a450d4802)

**Comment by sithlord48** [comment]
make more then one action for the hotkey
the dialog has a weird gui but you can have several actions mapped to one hotkey

**Comment by nbolton** [comment]
There are so many gaps that our current GUI doesn't cover.

Say you have a Linux client and Linux server... how does the user capture Meta+L on the server without locking it? Perhaps this has to be done with editing the config file? It would be nice if this was all done intuitively through the UI. How do you lock a Windows client remotely?

**Comment by nbolton** [comment]
If I map Ctrl+Alt+Ins on my Linux server to Ctrl+Meta+Q on my Mac, I have to move my mouse cursor to that Mac client to activate the hotkey for that client.

**Comment by elementalwindx** [comment]
Multiplicity can do synchronized unlock and locking. You have to put your windows password on the client pc (of course), but it's doable. I wish this had it as well. This is a pain in the butt to do.

**Comment by sithlord48** [comment]
Not a planed feature at all. But as a community project if you wanted to make Pr we would review it.
